### DISCUSSION TITLE
using-anonymous-types-and-tuples-to-attach-correlation-ids-to-scope-state-with-serilog-and-seq-in-asp-net-core/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#229] [Announcements]
# using-anonymous-types-and-tuples-to-attach-correlation-ids-to-scope-state-with-serilog-and-seq-in-asp-net-core/

In this post I describe two extension methods that make logging scopes as key-value pairs nicer in ASP.NET Core, by using anonymous types and tuples.

https://andrewlock.net/using-anonymous-types-and-tuples-to-attach-correlation-ids-to-scope-state-with-serilog-and-seq-in-asp-net-core/

[a comment by an automated account is not shown]

**Comment by georg138** [comment]
Thanks for this great article. Using collection expressions and a new `Enumerable.ToDictionary` overload from .NET8, there is a even more concise option from my point of view (which barely requires any additional code):
```cs
internal static class ILoggerExtensions
{
    public static IDisposable? BeginScope(this ILogger logger, IEnumerable<(string, object)> properties)
        => logger.BeginScope(properties.ToDictionary());
}
```
which can be used like (with on or more properties):
```cs
using var scope = _logger.BeginScope([("RecordingId", recordingId)]);
```
